src/Foundation/NSAction.cs

```diff
@@ -138,6 +155,57 @@ public void Apply ()
 				// MonoTouch has fixed the above problems, and we can call
 				// Dispose here.
 				Dispose ();
+#endif
+			}
+		}
+	}
+
+	// Use this for asynchronous operations
+	[Register ("__MonoMac_NSAsyncActionDispatcher2")]
+	internal class NSAsyncActionDispatcher2 : NSObject
```

Comment:
This class is almost entirely a copy-paste of `NSAsyncActionDispatcher`, and as such it seems it would be a good idea to introduce a base class for the common code.

Reply:
Sure! I can do that. I just thought about the benefits of not having virtual calls and co, but I can seal the classes. Will fix.